tests/session.py

```diff
@@ -3,44 +3,68 @@
 from sqlalchemy import create_engine
 
 from clickhouse_sqlalchemy import make_session
-from tests.config import uri, native_uri, system_native_uri
-
-session = make_session(create_engine(uri))
+from tests.config import (
+    http_uri, native_uri, system_native_uri, system_http_uri,
+)
 
+http_session = make_session(create_engine(http_uri))
+http_stream_session = make_session(create_engine(http_uri + '?stream=1'))
 native_session = make_session(create_engine(native_uri))
 
 system_native_session = make_session(create_engine(system_native_uri))
+system_http_session = make_session(create_engine(system_http_uri))
 
 
 class MockedEngine(object):
+
+    prev_do_execute = None
+    prev_do_executemany = None
+    prev_query_server_version_string = None
+
     def __init__(self, engine_session=None):
-        self.buffer = buffer = []
+        self._buffer = []
+
+        if engine_session is None:
+            uri = http_uri
+            engine_session = make_session(create_engine(
+                uri,
+                connect_args=dict(server_version='19.16.2.2')))
```

Comment:
Why we have to hardcode specific version here?

Reply:
This part is probably unnecessary anymore, in favor of the `query_server_version_string` override below

Reply:
Removed here, added a separate test for it.